src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourceArray = sourceString
+    .split(';')
+    .map((element) => element.trim());
+
+  const pairs = sourceArray
+    .map((element) => {
+      const [key, value] = element.split(':').map((part) => part.trim());
+      return key && value ? [key, value] : null;
+    })
+    .filter((pair) => pair);
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. The `forEach` loop in lines 22-24 could be replaced with a `reduce` function that directly builds the final object without the intermediate `pairs` array.